server/index.js

```diff
@@ -109,6 +113,54 @@ const getSalt = async event => {
   return { hashes: results }
 }
 
+/**
+ * Allows an authority to report an analysis
+ * @return {Promise<string>} Id of the report
+ */
+const reportAnalysis = async event => {
+  let authorization = event.headers['authorization']
+  if (!authorization) throw new ServerError("Missing header 'authorization'", 400)
+
+  let { date } = JSON.parse(event.body)
+  date = moment(date)
+  if (!date.isValid()) throw new ServerError("Missing or invalid 'date' prop.", 400)
+
+  const { db, client } = await getConnection()
+
+  // TODO: Improve authorization mechanism. Current weak implementation is for hackathon demo purposes
+  const valid = await dbAuthorities.validToken(db, authorization)
+  if(!valid) throw new ServerError("The authority is not valid'", 400)
+
+  const reportId = await dbAnalysisReports.saveReport(db, date)
+    .catch(() => {
+      throw new ServerError("Something went wrong reporting the analysis", 500)
+    })
+
+  client.close()
+
+  return reportId
+}
+
+/**
+ * Check the analysis report
+ * @return {Promise<string>} Analysis report
+ */
+const checkAnalysisReport = async event => {
+  const uuid = event.queryStringParameters && event.queryStringParameters['uuid']
+  if (!uuid) throw new ServerError("Missing 'uuid' prop.", 400)
+
+  const { db, client } = await getConnection()
+
+  const reportId = await dbAnalysisReports.getReport(db, uuid)
```

Comment:
Perhaps rename to `report` as I don't think this is an id here?
```suggestion
  const report = await dbAnalysisReports.getReport(db, uuid)
```

Reply:
Actually is the UUID of the report inserted in the database (not the complete report info), as we only want the endpoint to return the UUID to generate the QR. I will rename it to `reportUuid` 👍